### PR TITLE
Add option to push to gh-pages

### PR DESCRIPTION
**Pull request by unguiculus**
This adds the option to push the `index.yaml` file, either directly to the GitHub Pages branch, which is now configurable, or via pull request.

Signed-off-by: Reinhard Nägele <unguiculus@gmail.com>

### CODE REVIEW

**Comment by unguiculus** [comment]
I'd like to do a little refactoring in a follow-up PR in order to improve testability. With the current code, it is hard to add tests.

File: cr/cmd/index.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ package cmd
 
 import (
 	"github.com/helm/chart-releaser/pkg/config"
+	"github.com/helm/chart-releaser/pkg/git"
```
Comment (scottrigby) [inline cr/cmd/index.go:19]: @unguiculus what do you think about using https://github.com/go-git/go-git instead of our own package that shells out?

File: cr/cmd/index.go [hunk 2]
```diff
@@ -56,4 +57,7 @@ func init() {
 	flags.StringP("token", "t", "", "GitHub Auth Token (only needed for private repos)")
 	flags.StringP("git-base-url", "b", "https://api.github.com/", "GitHub Base URL (only needed for private GitHub)")
 	flags.StringP("git-upload-url", "u", "https://uploads.github.com/", "GitHub Upload URL (only needed for private GitHub)")
+	flags.String("pages-branch", "gh-pages", "The GitHub pages branch")
+	flags.Bool("push", false, "Push index.yaml to the GitHub Pages branch (must not be set if --pr is set)")
+	flags.Bool("pr", false, "Create a pull request for index.yaml against the GitHub Pages branch (must not be set if --push is set)")
```
Comment (scottrigby) [inline cr/cmd/index.go:62]: Nice 👍
Cross-referencing discussion in https://github.com/helm/chart-releaser-action/issues/33 about options design

File: pkg/releaser/releaser.go [hunk 3]
```diff
@@ -144,15 +162,62 @@ func (r *Releaser) UpdateIndexFile() (bool, error) {
...
+
+	if err := indexFile.WriteFile(r.config.IndexPath, 0644); err != nil {
+		return false, err
+	}
+
+	if !r.config.Push && !r.config.PR {
+		return true, nil
+	}
+
+	worktree, err := r.git.AddWorktree("", "origin/"+r.config.PagesBranch)
```
Comment (scottrigby) [inline pkg/releaser/releaser.go:181]: Don't want to scope creep too hard here, but we may now want to add a `remote` config option as well?

File: cr/cmd/index.go (hunk 1 above)
Comment (unguiculus) [reply cr/cmd/index.go:19]: We had used that in a client project and dropped it. We hit bugs and had various issues with it and it covers only a small subset of Git which wasn't enough for us. Plus, it's not really straight forward to use. I'd rather not use it.

Comment (unguiculus) [reply cr/cmd/index.go:19]: 'worktree' is not supported BTW.

File: pkg/releaser/releaser.go (hunk 3 above)
Comment (unguiculus) [reply pkg/releaser/releaser.go]: Good point. Added it.

File: cr/cmd/index.go (hunk 1 above)
Comment (scottrigby) [reply cr/cmd/index.go:19]: I saw that in the [compatibility doc](https://github.com/go-git/go-git/blob/master/COMPATIBILITY.md), but read through https://github.com/go-git/go-git/blob/master/worktree_test.go, which led me to think the doc just wasn't updated. But I don't have any real world experience using that package, so definitely trust your judgement 👍  My main thought was it could be cool to be able to run this without git as a dependency. Maybe something to reevaluate later?
